uri.go

```diff
@@ -68,11 +88,30 @@ func NewDocumentURI(path string) DocumentURI {
 	if len(path) == 0 || path[0] != '/' {
 		path = "/" + path
 	}
-	uri, err := NewDocumentURIFromURL("file://")
+	segments := strings.Split(path, "/")
+	encodedSegments := make([]string, len(segments))
+	for i, segment := range segments {
+		if len(segment) == 0 {
+			encodedSegments[i] = segment
+		} else {
+			segment = lowercaseDriveSegment(segment)
+			chars := strings.SplitAfter(segment, "")
+			for i, c := range chars {
+				// Spaces must be turned into `%20`. Otherwise, `url.QueryEscape`` encodes them to `+`.
+				if c == " " {
+					chars[i] = "%20"
+				} else {
+					chars[i] = url.QueryEscape(c)
+				}
+			}
+			encodedSegments[i] = strings.Join(chars, "")
```

Comment:
```suggestion
			segment = url.QueryEscape(segment)
			// Spaces must be turned into `%20`. Otherwise, `url.QueryEscape`` encodes them to `+`.
			encodedSegments[i] = strings.ReplaceAll(segment, "+", "%20")
```

Still not the best performance-wise, but at least we avoid the double-level split and join :-)